src/ORM/Connect/DBSchemaManager.php

```diff
@@ -593,12 +558,12 @@ protected function parseIndexSpec($name, $spec)
     protected function convertIndexSpec($indexSpec)
     {
         // Return already converted spec
-        if (!is_array($indexSpec)) {
-            return $indexSpec;
+        if (!is_array($indexSpec) || !array_key_exists('type', $indexSpec) || !array_key_exists('columns', $indexSpec) || !is_array($indexSpec['columns'])) {
```

Comment:
As it's written `DataObjectSchema` is in charge of normalising the indexes. There's no way that the index spec should be loosely defined by the time it's made its way to the `DBSchemaManager`.

If you don't define a type in your config, `DataObjectSchema` assumes it's 'index'. The `DBSchemaManager` is strict with what it accepts but we are still loose with what we can put in the config.